src/librustc_trans/collector.rs

```diff
@@ -850,8 +910,14 @@ impl<'b, 'a, 'v> ItemLikeVisitor<'v> for RootCollector<'b, 'a, 'v> {
                 // const items only generate translation items if they are
                 // actually used somewhere. Just declaring them is insufficient.
             }
-            hir::ItemFn(.., ref generics, _) => {
-                if !generics.is_type_parameterized() {
+            hir::ItemFn(_, _, constness, _, ref generics, _) => {
+                let is_const = match constness {
```

Comment:
Does `is_const_fn` do something differently than the code below? I'm not against changing, but the `constness` and `generics` from the HIR item are right there without the need for a hashtable lookup.